bundles/com.espressif.idf.core/src/com/espressif/idf/core/util/ClangdConfigFileHandler.java

```diff
@@ -35,33 +36,50 @@ public class ClangdConfigFileHandler
 	public void update(IProject project) throws CoreException, IOException
 	{
 		File file = getClangdConfigFile(project);
+		FileInputStream inputStream = null;
```

Comment:
Since FileInputStream is AutoClosable we can use try-with-resources. Also, to reduce nesting we can put both inputStream and fileWriter into one try-with-resources block like this:
```
try (FileInputStream inputStream = new FileInputStream(file); Writer writer = new FileWriter(file))
{}
```

Reply:
fixed